dev-proxy/InactivityTimer.cs

```diff
@@ -0,0 +1,30 @@
+using DevProxy.Abstractions;
+
+namespace DevProxy
+{
+    public class InactivityTimer : IInactivityTimer
+    {
+        private Timer? _timer;
+        private readonly TimeSpan _timeout;
+        private static readonly TimeSpan InfiniteTimeout = TimeSpan.FromMilliseconds(-1);
+
+        public InactivityTimer(long timeoutSeconds, Action timeoutAction)
+        {
+            _timeout = TimeSpan.FromSeconds(timeoutSeconds);
+            Action action = timeoutAction ?? throw new ArgumentNullException(nameof(timeoutAction));
+        
+            _timer = new Timer(_ => action.Invoke(), null, _timeout, InfiniteTimeout);
```

Comment:
For the InfiniteTimeout, let's use `System.Threading.Timeout.InfiniteTimeSpan` rather than declaring our own property.